zulip/zulip/__init__.py

```diff
@@ -106,6 +107,12 @@ def _default_client():
 
 def add_default_arguments(parser):
     # type: (argparse.ArgumentParser) ->  argparse.ArgumentParser
+
+    def custom_error_handling(self, message):
+        self.print_help(sys.stderr)
+        self.exit(2, '{prog}: error: {message}\n'.format(self.prog, message))
+    parser.error = types.MethodType(custom_error_handling, parser)
```

Comment:
I would make this behavior be controlled by a kwarg, e.g. if you specify `add_default_arguments(parser, patch_error_handling=False)`, it won't do this.  Because I think this done right will be better than the default for most users.

That said, I think the actual output of this version is busted:

```
$ ./zulip/examples/send-message --asdf
...
  --client-cert-key CLIENT_CERT_KEY
                        Specify a file containing the client certificate's key
                        (if it is in a separate file).
Traceback (most recent call last):
  File "./zulip/examples/send-message", line 41, in <module>
    options = parser.parse_args()
  File "/usr/lib/python2.7/argparse.py", line 1704, in parse_args
    self.error(msg % ' '.join(argv))
  File "/home/tabbott/python-zulip-api/zulip/zulip/__init__.py", line 113, in custom_error_handling
    self.exit(2, '{prog}: error: {message}\n'.format(self.prog, message))
KeyError: 'prog'
```